src/Microsoft.AspNet.OData.Shared/DeltaOfTStructuralType.cs

```diff
@@ -121,6 +169,21 @@ public override bool TrySetPropertyValue(string name, object value)
                 throw Error.ArgumentNull("name");
             }
 
+            if (_instanceAnnotationsPropertyInfo != null)
+            {                
+                if (name == _instanceAnnotationsPropertyInfo.Name)
+                {
+                    IODataInstanceAnnotationContainer annotationValue = value as IODataInstanceAnnotationContainer;
+                    if (_instanceAnnotationCache == null && annotationValue != null)
```

Comment:
>annotationValue != null [](start = 60, length = 23)

Should we not allow them to set the value of the _instanceAnnotationCache to null?  (we should return false if they try to set a non-null value that is not an IODataInstanceAnnotationContainer, but we could let them pass null for value)